src/engine/TPA.cc

```diff
@@ -73,8 +74,17 @@ VerificationResult TPAEngine::solve(const ChcDirectedGraph & graph) {
                 assert(false);
                 throw std::logic_error("Unreachable!");
         }
-    } else if (isTransitionSystemDAG(graph) && not options.hasOption(Options::FORCE_TS)) {
-        return solveTransitionSystemGraph(graph);
+    } else if ((isTransitionSystemDAG(graph) && not options.hasOption(Options::FORCE_TS)) ||
+               options.hasOption(Options::SIMPLIFY_NESTED)) {
+        if (options.hasOption(Options::SIMPLIFY_NESTED)) {
+            NestedLoopTransformation transformation;
+            auto normalGraph = graph;
+            auto preTranslator = transformation.transform(normalGraph);
+            auto res = solveTransitionSystemGraph(normalGraph);
```

Comment:
Add an `assert` that this `isTransitionSystemGraph` now.

Reply:
Fixed